planner/logical_test.go

```diff
@@ -0,0 +1,180 @@
+package planner_test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/functions"
+	"github.com/influxdata/flux/planner"
+	"github.com/influxdata/flux/planner/plannertest"
+	"github.com/influxdata/flux/semantic"
+)
+
+var create = map[flux.OperationKind]planner.CreateLogicalProcedureSpec{
+	// Take a FromOpSpec and translate it to a FromProcedureSpec
+	functions.FromKind: func(op flux.OperationSpec) (planner.LogicalProcedureSpec, error) {
+		spec, ok := op.(*functions.FromOpSpec)
+
+		if !ok {
+			return nil, fmt.Errorf("invalid spec type %T", op)
+		}
+
+		return &planner.FromProcedureSpec{
+			Bucket:   spec.Bucket,
+			BucketID: spec.BucketID,
+		}, nil
+	},
+	// Take a RangeOpSpec and convert it to a RangeProcedureSpec
+	functions.RangeKind: func(op flux.OperationSpec) (planner.LogicalProcedureSpec, error) {
+		spec, ok := op.(*functions.RangeOpSpec)
+
+		if !ok {
+			return nil, fmt.Errorf("invalid spec type %T", op)
+		}
+
+		if spec.TimeCol == "" {
+			spec.TimeCol = execute.DefaultTimeColLabel
+		}
+
+		return &planner.RangeProcedureSpec{
+			Bounds: flux.Bounds{
+				Start: spec.Start,
+				Stop:  spec.Stop,
+			},
+			TimeCol:  spec.TimeCol,
+			StartCol: spec.StartCol,
+			StopCol:  spec.StopCol,
+		}, nil
+	},
+	// Take a FilterOpSpec and translate it to a FilterProcedureSpec
+	functions.FilterKind: func(op flux.OperationSpec) (planner.LogicalProcedureSpec, error) {
+		spec, ok := op.(*functions.FilterOpSpec)
+
+		if !ok {
+			return nil, fmt.Errorf("invalid spec type %T", op)
+		}
+
+		return &planner.FilterProcedureSpec{
+			Fn: spec.Fn.Copy().(*semantic.FunctionExpression),
+		}, nil
+	},
+}
+
+func TestFluxSpecToLogicalPlan(t *testing.T) {
+	testcases := []struct {
+		name string
+		spec *flux.Spec
```

Comment:
How about starting from Flux text, translating it to a Spec and then to the logical plan in the `Run` method below?  That way these tests would be somewhat easier to read.

Reply:
That's true. Would these still be considered unit tests if we were to do that? Does is matter if they're not unit tests?

Reply:
Hmm.  That's a good question.  Personally, I think it's okay to start from a Flux program in unit tests, because it's easier to read, and it provides a convenient way to produce a Spec whose graph has the topology you want to test.  It's pretty much a 1:1 conversion from Spec operations to logical nodes, so it's not adding a whole lot of complexity.